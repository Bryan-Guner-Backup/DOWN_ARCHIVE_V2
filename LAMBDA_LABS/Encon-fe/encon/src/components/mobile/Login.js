import React, { useState } from 'react';
import axios from 'axios';
import '../../styles/mobile/Login.scss';
import { useForm } from 'react-hook-form';
import { HeaderAlt } from './Header-Alt.js';
import { useHistory } from 'react-router-dom';
import { ErrorMessage } from './Error-Message.js';

export const Login = () => {
  const { handleSubmit, register, errors, reset } = useForm();
  const baseUrl = 'https://encon-be.herokuapp.com/api';
  // const [loginError, setLoginError] = useState();
  const history = useHistory();

  const onLoginSubmit = (data) => {
    axios
      .post(baseUrl + '/auth/login', {
        email: data.email,
        password: data.password,
      })
      .then((res) => {
        reset();
        localStorage.setItem('AUTH_TOKEN', res.data.token);
        localStorage.setItem('USER_ID', res.data.Data.id);
        localStorage.setItem('USER_NAME', res.data.Data.name);
        localStorage.setItem('USER_LOCATION', res.data.Data.state);
        history.push('/profile');
      })
      .catch((err) => {
        // setLoginError('Login Error: ' + err.response.data.error.message);
      });
  };

  return (
    <div className='login-container'>
      <HeaderAlt />
      <form className='loginForm' onSubmit={handleSubmit(onLoginSubmit)}>
        <label htmlFor='email' className='label'>
          Email
        </label>
        <input
          type='email'
          name='email'
          ref={register({ required: true, pattern: /^\S+@\S+$/i })}
        />
        <ErrorMessage error={errors.email} />
        {errors.email && errors.email.message}
        <label htmlFor='password' className='label'>
          Password
        </label>
        <input
          type='password'
          name='password'
          ref={register({ required: true })}
        />
        <ErrorMessage error={errors.password} />
        <button className='app-buttons' type='submit' data-testid='sign in'>
          Sign In
        </button>
        {/* <div>{loginError}</div> */}
      </form>
    </div>
  );
};
