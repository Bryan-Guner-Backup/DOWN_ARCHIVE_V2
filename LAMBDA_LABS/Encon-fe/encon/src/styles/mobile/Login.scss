.login-container {
	margin: 3rem 1.75rem;

	.header-alt-container {
		margin: 3.25rem 0;
	}

	.label {
		display: flex;
		font-size: 1.35rem;
		font-weight: 500;
		margin-top: 1.75rem;
		padding-bottom: 0.75rem;
	}

	input {
		font-size: 1.5rem;
		width: 100%;
		padding: 1.05rem 1.75rem;
		border-radius: 1.25rem;
		border: 0.5px solid transparentize($color: #4f4f4f, $amount: 0.75);
	}
}
