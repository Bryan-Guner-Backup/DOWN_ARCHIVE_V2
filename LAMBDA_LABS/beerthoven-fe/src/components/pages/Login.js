import React from  "react";
import { useForm } from 'react-hook-form';
import { Form, Input, Button } from 'antd';

function Login() {
  const { register, handleSubmit, errors } = useForm()
  const onSubmit = data => { console.log("Here's Johnny!", data) }  

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>      
      <Input
        type='text'
        placeholder='Email'      
        ref={register({ required: true })} />
      <Input 
      type='text'
      placeholder='Password'      
      ref={register({ required: true })} />
      {errors.userName && <span>This field is required</span>}
      {errors.password && <span>This field is required</span>}
      <Button type='default'>Login</Button>   
    </Form>
  )
  
}

export default Login;