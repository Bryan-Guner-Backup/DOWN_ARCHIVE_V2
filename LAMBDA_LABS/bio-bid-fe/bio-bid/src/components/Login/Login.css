.LoginLink {
  margin-left: 1%;
}

Button.Logout {
  margin-left: 2%;
  background-color: #F5222D;
  color: white;
  text-transform: none;
  font-family: Inter, sans-serif;
  width: 100px;
}
Button.Logout:hover {
  color: #096dd9;
  background-color: white;
}

.LoginLink:hover {
  opacity: 0.8;
}

.LoginLink img {
  width: 200px;
}
