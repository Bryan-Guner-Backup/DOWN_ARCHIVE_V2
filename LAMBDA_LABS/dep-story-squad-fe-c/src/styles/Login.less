.loginarea {
  display: flex;
  flex-wrap: wrap;

  .textarealogin {
    width: 100%;
    margin: 3%;
    padding: 0 3%;
    line-height: 1.5;
    font-size: 1rem;
  }

  //okta login widget ---------------------------
  #sign-in-widget {
    width: 100%;
    margin: 0 auto;

    .auth-header {
      display: none;
    }

    .okta-form-title {
      text-align: center;
      font-size: 24px;
      line-height: 22px;
      color: rgba(60, 60, 67, 0.6);
    }

    .o-form-fieldset-container {
      text-align: center;
    }

    .o-form-button-bar {
      margin-top: 6%;
      text-align: center;
      margin-bottom: 5%;
    }

    #okta-signin-username {
      appearance: none;
      width: 70%;
    }

    #okta-signin-password {
      width: 70%;
    }

    .js-help {
      padding-left: 35%;
      color: black;
    }
  }
  //okta login widget ---------------------------

  // Media Query for desktop
  .MQ(
    @desktop,
    {flex-direction: row-reverse; .textarealogin {width: 50%; display: flex;
      flex-wrap: wrap; align-content: center; font-size: 1.25rem;} #sign-in-widget
      {width: 40%; .auth-container {margin: 5%;}}}
  );
}
