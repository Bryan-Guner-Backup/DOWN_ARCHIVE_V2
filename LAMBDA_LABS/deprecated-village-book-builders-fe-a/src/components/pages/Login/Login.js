import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { Row, Col, Form, Input, Button, Alert } from 'antd';
import axios from 'axios';
import Password from 'antd/lib/input/Password';
import { useUser } from '../../../state/UserContext';

function Login() {
  const user = useUser();
  let history = useHistory();
  const [form] = Form.useForm();
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (user.userInfo.role === 'admin') {
      history.push('/admin/library');
    } else if (user.userInfo.role === 'headmaster') {
      history.push('/dashboard');
    }
  }, []);

  const onFinish = values => {
    setLoading(true);
    axios
      .get(`https://54.158.134.245/api/auth/?username=${values.username}`)
      .then(res => {
        setLoading(false);
        user.setUserInfo({
          username: res.data[0].username,
          role: res.data[0].role,
          id: res.data[0].id,
        });
        if (res.data[0].role === 'admin') {
          history.push('/admin/library');
        } else if (res.data[0].role === 'headmaster') {
          history.push('/dashboard');
        }
      })
      .catch(err => {
        setLoading(false);
        setError(err);
      });
  };

  const onFinishFailed = errorInfo => {
    setError(errorInfo);
  };

  return (
    <Row>
      <Col span={10} offset={8}>
        <h1>Village Book Builders Login</h1>

        {error && (
          <Alert
            message="There was an error"
            description="Your request could not be completed"
            type="error"
            closable
          />
        )}
        <Form
          form={form}
          layout="vertical"
          initialValues={{
            remember: true,
          }}
          onFinish={onFinish}
          onFinishFailed={onFinishFailed}
        >
          <Form.Item
            label="Username"
            name="username"
            rules={[{ required: true }]}
          >
            <Input />
          </Form.Item>

          <Form.Item
            label="Password"
            name="password"
            rules={[{ required: true }]}
          >
            <Password />
          </Form.Item>

          <Form.Item>
            <Button
              shape="round"
              type="primary"
              htmlType="submit"
              loading={loading}
            >
              Login
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </Row>
  );
}

export default Login;
