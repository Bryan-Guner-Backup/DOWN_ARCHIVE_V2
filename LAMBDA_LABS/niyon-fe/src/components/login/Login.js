import React from 'react'
import { withFormik, Form, Field } from 'formik'
import * as Yup from 'yup'
import { logIn } from '../apiStuff/axiosWithAuth'
import Header from '../header/Header'

const Login = (props) => {
  const { values, errors, touched, isSubmitting } = props

  const handleSubmit = (event) => {
    /*eslint-disable */
    const login = { email: values.email, password: values.password };
    event.preventDefault();   
    logIn(values).then((res) => {
        window.localStorage.setItem("token", res.data.token);
        window.localStorage.setItem("id", res.data.user.user_id);
        window.localStorage.setItem("user_type", res.data.user.user_type);
        window.location = "/home";
      })
      .catch((err) => {
        console.log(err);
      });
  };

  return (
    <div className="formLogin">
      <Header />
      <Form onSubmit={handleSubmit} data-test="form">
        <div className="inputWrap">
          <div>
            {touched.username && errors.username && <p>{errors.username}</p>}
            <Field type="email" name="email" placeholder="Email" />
          </div>
          <div>
            {touched.password && errors.password && <p>{errors.password}</p>}
            <Field type="password" name="password" placeholder="Password" />
          </div>
          <br />
          <button className="button" disabled={isSubmitting} data-test="submit">
            Login
          </button>
        </div>
      </Form>
    </div>
  );
};

const FormikLogin = withFormik({
  mapPropsToValues({ email, password }) {
    return {
      email: email || "",
      password: password || "",
    };
  },
  validationSchema: Yup.object().shape({
    email: Yup.string().required("Email is required"),
    password: Yup.string()
      .min(6, "Password must be 6 chracters or longer")
      .required("Password is required"),
  }),
})(Login);

export default FormikLogin;
