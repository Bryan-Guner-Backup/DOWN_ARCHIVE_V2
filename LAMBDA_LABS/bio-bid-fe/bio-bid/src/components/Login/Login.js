import React from "react";
import "./Login.css";
import SignIn from "../../images/Sign-In-Small.png";
import { useOktaAuth } from "@okta/okta-react";
import AutoLogin from "./AutoLogin";
import { Button } from "@material-ui/core";

export default function Login() {
  const { authState, authService } = useOktaAuth();
  // console.log({ authState, authService });
  const linkedin = {
    url: "https://dev-648803.okta.com/oauth2/v1/authorize",
    idp: "0oadk2c9qgPCXavKf4x6",
    clientId: "0oadk3f256MfKqlA74x6",
    responseType: "id_token token",
    responseMode: "fragment",
    scope: "openid%20profile%20email",
    redirectUri: `${window.location.origin}`,
    state: "ANYVALUE",
    nonce: "ANYVALUE",
  };
  const {
    url,
    idp,
    clientId,
    responseType,
    responseMode,
    scope,
    redirectUri,
    state,
    nonce,
  } = linkedin;

  const finalUrl = `${url}?idp=${idp}&client_id=${clientId}&response_type=${responseType}&response_mode=${responseMode}&scope=${scope}&redirect_uri=${redirectUri}&state=${state}&nonce=${nonce}`;

  const logout = () => {
    authService.logout("/"); // Redirect to '/' after logout
  };

  if (authState.isPending) {
    return (
      <div>
        <AutoLogin />
      </div>
    );
  }

  return authState.isAuthenticated ? (
    <Button
      className="Logout"
      variant="contained"
      color="primary"
      size="small"
      onClick={logout}
    >
      Sign Out
    </Button>
  ) : (
    <a className="LoginLink" href={finalUrl}>
      <img alt="LinkedIn Logo Login Button" src={SignIn}></img>
    </a>
  );
}
