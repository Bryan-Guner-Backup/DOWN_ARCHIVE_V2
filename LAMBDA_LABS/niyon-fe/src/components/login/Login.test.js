import React from 'react';
import { shallow,mount } from 'enzyme';
import { Form }from 'formik';
import toJSON from 'enzyme-to-json';
import Login from './Login';



describe('<Login /> component testing', () => {
  let component
  let onMockSubmit
  beforeEach(() => {
    onMockSubmit = () => jest.fn()
    component = shallow(<Login onSubmit={onMockSubmit} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should pass snaps shot testing', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render <Login /> component correctly', () => {   
    expect(component.exists()).toBe(true)
  })

  it('should render form correctly', () => {
    // console.log(component.dive().dive().debug())
    expect(component.find(Form)).toBeTruthy()
  })

 it('should render the input element for email field', () => {
    expect(component.find('input[name="email"]')).toBeTruthy()
  })


  it('should render the input element for password field', () => {
    expect(component.find('input[name="password"]')).toBeTruthy()
  })

  it('should render the submit button correctly', () => {
    expect(component.find('[data-test="submit"]')).toBeTruthy()
  })

  it('it should submit the form when user clicks the submit button', () => {
    const form = mount(<Login/>)
       const submitButton = form.find(".button")
       submitButton.simulate('click')
      
  })
  
})
