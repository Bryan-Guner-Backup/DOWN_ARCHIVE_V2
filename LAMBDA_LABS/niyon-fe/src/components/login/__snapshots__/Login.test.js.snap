// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component testing should pass snaps shot testing 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={[Function]}
>
  <Component />
</Formik>
`;
